### DISCUSSION TITLE
built rtags

### DISCUSSION THREAD

**Discussion by xiaojbing** [idaholab/moose#19594] [Q&A Getting Started] [answered]
I try to build rtags in my ubantu 20.04，

git clone https://github.com/Andersbakken/rtags.git
cd rtags/
git submodule init
git submodule update
mkdir mybuild && cd mybuild
cmake ..
make
but when I use the command :make ,an error appeared as followed.

[  1%] Building CXX object src/CMakeFiles/rtags.dir/ClangIndexer.cpp.o
[  2%] Building CXX object src/CMakeFiles/rtags.dir/ClangThread.cpp.o
[  4%] Building CXX object src/CMakeFiles/rtags.dir/ClassHierarchyJob.cpp.o
[  5%] Building CXX object src/CMakeFiles/rtags.dir/CompilerManager.cpp.o
[  7%] Building CXX object src/CMakeFiles/rtags.dir/CompletionThread.cpp.o
[  8%] Building CXX object src/CMakeFiles/rtags.dir/DependenciesJob.cpp.o
[  9%] Building CXX object src/CMakeFiles/rtags.dir/IncludePathJob.cpp.o
[ 11%] Building CXX object src/CMakeFiles/rtags.dir/FileManager.cpp.o
[ 12%] Building CXX object src/CMakeFiles/rtags.dir/FindFileJob.cpp.o
[ 14%] Building CXX object src/CMakeFiles/rtags.dir/FindSymbolsJob.cpp.o
[ 15%] Building CXX object src/CMakeFiles/rtags.dir/FollowLocationJob.cpp.o
[ 16%] Building CXX object src/CMakeFiles/rtags.dir/IncludeFileJob.cpp.o
[ 18%] Building CXX object src/CMakeFiles/rtags.dir/IndexMessage.cpp.o
[ 19%] Building CXX object src/CMakeFiles/rtags.dir/IndexParseData.cpp.o
[ 21%] Building CXX object src/CMakeFiles/rtags.dir/IndexerJob.cpp.o
[ 22%] Building CXX object src/CMakeFiles/rtags.dir/JobScheduler.cpp.o
[ 23%] Building CXX object src/CMakeFiles/rtags.dir/ListSymbolsJob.cpp.o
[ 25%] Building CXX object src/CMakeFiles/rtags.dir/Location.cpp.o
[ 26%] Building CXX object src/CMakeFiles/rtags.dir/Preprocessor.cpp.o
[ 28%] Building CXX object src/CMakeFiles/rtags.dir/Project.cpp.o
[ 29%] Building CXX object src/CMakeFiles/rtags.dir/QueryJob.cpp.o
[ 30%] Building CXX object src/CMakeFiles/rtags.dir/QueryMessage.cpp.o
[ 32%] Building CXX object src/CMakeFiles/rtags.dir/RClient.cpp.o
[ 33%] Building CXX object src/CMakeFiles/rtags.dir/RTags.cpp.o
[ 35%] Building CXX object src/CMakeFiles/rtags.dir/ReferencesJob.cpp.o
[ 36%] Building CXX object src/CMakeFiles/rtags.dir/Sandbox.cpp.o
[ 38%] Building CXX object src/CMakeFiles/rtags.dir/ScanThread.cpp.o
[ 39%] Building CXX object src/CMakeFiles/rtags.dir/Server.cpp.o
[ 40%] Building CXX object src/CMakeFiles/rtags.dir/ServerMessageHandlers.cpp.o
[ 42%] Building CXX object src/CMakeFiles/rtags.dir/Source.cpp.o
[ 43%] Building CXX object src/CMakeFiles/rtags.dir/StatusJob.cpp.o
[ 45%] Building CXX object src/CMakeFiles/rtags.dir/Symbol.cpp.o
[ 46%] Building CXX object src/CMakeFiles/rtags.dir/SymbolInfoJob.cpp.o
[ 47%] Building CXX object src/CMakeFiles/rtags.dir/Token.cpp.o
[ 49%] Building CXX object src/CMakeFiles/rtags.dir/TokensJob.cpp.o
[ 50%] Linking CXX static library librtags.a
[ 50%] Built target rtags
Scanning dependencies of target rct
[ 52%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Buffer.cpp.o
[ 53%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Config.cpp.o
[ 54%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Connection.cpp.o
[ 56%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/CpuUsage.cpp.o
[ 57%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Date.cpp.o
[ 59%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/EventLoop.cpp.o
[ 60%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/FileSystemWatcher.cpp.o
[ 61%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Log.cpp.o
[ 63%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/MemoryMonitor.cpp.o
[ 64%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Message.cpp.o
[ 66%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/MessageQueue.cpp.o
[ 67%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Path.cpp.o
[ 69%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Plugin.cpp.o
[ 70%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Rct.cpp.o
[ 71%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/ReadWriteLock.cpp.o
[ 73%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/Semaphore.cpp.o
[ 74%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/SharedMemory.cpp.o
[ 76%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/SocketClient.cpp.o
[ 77%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/SocketServer.cpp.o
[ 78%] Building CXX object src/CMakeFiles/rct.dir/rct/rct/String.cpp.o
/home/xjb/rtags/src/rct/rct/String.cpp:3:10: fatal error: zconf.h:No such file or directory
    3 | #include <zconf.h>
      |          ^~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/rct.dir/build.make:310：src/CMakeFiles/rct.dir/rct/rct/String.cpp.o] error 1
make[1]: *** [CMakeFiles/Makefile2:1110：src/CMakeFiles/rct.dir/all] error 2
make: *** [Makefile:141：all] error 2

so, how should I solve this ?

**Comment by GiudGiud** [comment, marked as answer]
See this answer
https://stackoverflow.com/questions/67621721/zlib-h3419-fatal-error-zconf-h-no-such-file-or-directory-include-zconf-h

seems you need to install a package for it

**Reply by xiaojbing** [reply]
thanks very much , it works. but when I run

`MOOSE_UNITY=false make compile_commands.json && rc -J .`

in the directory **projects/moose/modules**, a new error occurred.

```
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/libtool: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/libtool: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
Using HIT from /home/xjb/projects/moose/framework/contrib/hit
/bin/sh: 1: /home/xjb/projects/moose/libmesh/installed/contrib/bin/libmesh-config: not found
[Server] Compilation failed to load.

```

**Reply by xiaojbing** [reply]
I forget to use 

`conda activate moose`

sorry. . .   there is no problem now.

**Reply by GiudGiud** [reply]
nice glad it works!
